Controller/Payment/Webhook.php

```diff
@@ -569,18 +565,39 @@ protected function setWebhookNotifiedAt($entity_id)
         $order->save();
     }
 
-    protected function setWebhookData($entityId, $webhookVerifiedStatus, $paymentId, $event, $amount)
+    protected function setWebhookData($post, $entityId, $webhookVerifiedStatus, $paymentId, $amount)
     {
+        $order                  = $this->order->load($entityId);
+        $existingWebhookData    = $order->getRzpWebhookData();
+
+        if ($post['event'] === 'payment.authorized')
+        {
+            $amount = $post['payload']['payment']['entity']['amount'];
+        }
+        else if ($post['event'] === 'order.paid')
+        {
+            $amount = $post['payload']['order']['entity']['amount_paid'];
+        }
         $webhookData = array(
             "webhook_verified_status"   => $webhookVerifiedStatus,
             "payment_id"                => $paymentId,
-            "event"                     => $event,
             "amount"                    => $amount 
         );
-        
-        $webhookDataText = serialize($webhookData);
 
-        $order = $this->order->load($entityId);
+        if (!empty($existingWebhookData))
+        {
+            $existingWebhookData = unserialize($existingWebhookData);
+            if (!array_key_exists($post['event'], $existingWebhookData))
+            {
+                $existingWebhookData[$post['event']] = $webhookData;
+            }
```

Comment:
@nikhilsharma1100 Please add new line after this.

Reply:
done